### PR TITLE
Add CTC recipe to AISHELL-1

### PR DESCRIPTION
**Pull request by BenoitWang**
Hi @mravanelli @TParcollet , this PR adds a typical CTC-wav2vec recipe to AISHELL-1.
Test CER: 5.06%
Dev CER: 4.52%

Some points:
1. [chinese-wav2vec2-large](https://huggingface.co/TencentGameMate/chinese-wav2vec2-large) (from Tencent) is used which is pretrained on 10k hours Chinese data
2. [bert-base-chinese](https://huggingface.co/bert-base-chinese) is used as the tokenizer, ctc is trained on chars
3. In `prepare.py`, `pandas` is not necessary to be used to generate csv, so it is deleted together with some unused variables.

### CODE REVIEW

**Comment by TParcollet** [comment]
Huge ! Is this comparable to the SOTA around?

**Comment by BenoitWang** [comment]
Hi @TParcollet , I think it's good for a system pure-CTC/greedy/without LM. 
Hybrid models from espnet got better CER:

model | Test CER | Dev CER | LM
-- | -- | -- | --
our ctc-wav2vec | 5.06% | 4.52% | No
espnet: branchformer-beam10-ctc0.4 | 4.4% | 4.1% | No
espnet: conformer-beam20-ctc0.3 | 4.9% | 4.5% | No

**Comment by TParcollet** [comment]
I see, not bad, but we use extra pre-training while they don't, correct ?

**Comment by BenoitWang** [comment]
Yes exact. Fair enough, but their [branchformer](https://arxiv.org/pdf/2207.02971.pdf) is quite something according to the results.

**Review by anautsch — CHANGES_REQUESTED** [review]
Hi @BenoitWang minor details only.

The yaml file combines related hparam files well; as is for the train script.

Is the AISHELL-1 prepare script completely stripped of the extra dependency to `pandas` for all its recipes? 
(not a bad thing to reduce dependencies, although pandas is neat, - just asking - pandas is not in the SB requirements and neither it is explicitly stated fo AISHELL-1, so it's a good catch)

**Comment by BenoitWang** [comment]
Hi @anautsch thanks for the review, the fix is done. And yes that's why I want to reduce pandas, it is only used to generate csv for all the recipes.

**Review by anautsch — APPROVED** [review]

File: recipes/AISHELL-1/ASR/CTC/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+If not present in the specified data_folder, the dataset will be automatically downloaded there.
+This step is not mandatory. We will use the official tokenizer downloaded from the web if you do not
+specify a different tokenizer in the speech recognition recipe.
+
+2- Train the speech recognizer
+```
+python train_with_wav2vec.py hparams/train_with_wav2vec.yaml
+```
+
+Make sure to have "transformers" installed.
```
Comment (TParcollet) [inline recipes/AISHELL-1/ASR/CTC/README.md:18]: please add an extra_requirements.txt file with transformer in it :-)

File: recipes/AISHELL-1/ASR/CTC/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+You can checkout our results (models, training logs, etc,) here:
+To be added
+
+# Training Time
+It takes about 2h on 1 RTX 8000 (48GB)
+
+# PreTrained Model + Easy-Inference
+You can find the pre-trained model with an easy-inference function on HuggingFace
+- To be added
```
Comment (TParcollet) [inline recipes/AISHELL-1/ASR/CTC/README.md:35]: Must be added before merging, I guess?

File: recipes/AISHELL-1/ASR/CTC/prepare.py [hunk 3]
```diff
@@ -0,0 +1 @@
+../../prepare.py
```
Comment (TParcollet) [inline recipes/AISHELL-1/ASR/CTC/prepare.py:1]: Hey, the naming convention of the file is not equivalent to other recipes, please have a look at the name and just rename the file accordingly.

File: recipes/AISHELL-1/ASR/CTC/train_with_wav2vec.py [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+    @sb.utils.data_pipeline.takes("transcript")
+    @sb.utils.data_pipeline.provides(
+        "wrd", "tokens_list", "tokens_bos", "tokens_eos", "tokens"
+    )
+    def text_pipeline(wrd):
+        wrd = "".join(wrd.split(" "))
+        yield wrd
+        tokens_list = tokenizer(wrd)["input_ids"]
+        yield tokens_list
+        tokens_bos = torch.LongTensor([hparams["bos_index"]] + (tokens_list))
```
Comment (TParcollet) [inline recipes/AISHELL-1/ASR/CTC/train_with_wav2vec.py:260]: If not mistaken, bos and Eos tokens are not used, they could be removed from the text_pipeline.

**Review by anautsch — APPROVED** [review]

**Comment by BenoitWang** [comment]
Hi @TParcollet @anautsch @Adel-Moumen,

Thank you all for the reviews and tests! The HF link is added, here's a brief summary of the PR:
1. add a CTC recipe
2. fix naming problems
3. fix dynamic batching conflicts for seq2seq & transformer recipes

**Comment by anautsch** [comment]
lgtm.

Tested recipes in `--debug` mode & the wav2vec2 with ddp.

---

Side note: we have an internal issue with `--debug` and eval checkpointing - this comes clear when running this transformer wav2vec2 recipe - here's the relevant log
```
   asr_brain.evaluate(
  File "speechbrain/core.py", line 1260, in evaluate
    self.on_evaluate_start(max_key=max_key, min_key=min_key)
  File "train_with_wav2vect.py", line 272, in on_evaluate_start
    ckpt = sb.utils.checkpoints.average_checkpoints(
  File "speechbrain/utils/checkpoints.py", line 1174, in average_checkpoints
    return averager(parameter_iterator)
  File "speechbrain/utils/checkpoints.py", line 1080, in average_state_dicts
    raise ValueError("No state dicts to average.")
ValueError: No state dicts to average.
```

**Pull request merged**